xarray/core/parallel.py

```diff
@@ -25,12 +27,29 @@
 
 import numpy as np
 
+from .alignment import align
 from .dataarray import DataArray
 from .dataset import Dataset
 
 T_DSorDA = TypeVar("T_DSorDA", DataArray, Dataset)
 
 
+def to_object_array(iterable):
+    npargs = np.empty((len(iterable),), dtype=np.object)
```

Comment:
This could be written as `np.empty_like(iterable, dtype=np.object)`.

Reply:
`empty_like` computes things.